### DISCUSSION TITLE
How to enumerate elements of FieldTable

### DISCUSSION THREAD

**Discussion by alllel** [gftea/amqprs#95] [Q&A] [answered]
Is there a way to list/enumerate/iterate over keys in FieldTable?
You see: FieldTable is necessary not only to send messages, but it also belongs to received messages.
And sometimes it would be nice to inspect it in detail.

I see that internally there is just a HashMap, but it is not public.

What if to add to FieldTable something like: 
```
impl FieldTable {
  pub fn keys(&self) {
   self.1.keys()
  }
}
```

**Comment by gftea** [comment, marked as answer]
We can implement AsRef, something like below, would you like to submit a PR for this?
```rust
impl AsRef<HashMap<FieldName, FieldValue>> for FieldTable {
   fn as_ref(&self) -> &HashMap<FieldName, FieldValue> {
}
}
```
